Sign up

llabake opened the pull request:
# Add user sign up functionality
-  configure our production application to use SSL for secure communications and Puma for high performance.
- interact with users as a resource through a standard set of REST URLs.
- display temporary messages using `flash`

llabake committed:
Finish user signup

llabake committed:
Use SSL and the Puma webserver in production

chalchuck commented on app/controllers/users_controller.rb:
```diff
@@ -1,4 +1,25 @@
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = "Welcome to the Sample App!"
```
combine this to be a one liner and always use single quotes if you are not using interpolation.

```suggestion
      redirect_to @user, notice: 'Welcome to the Sample App!'
```

chalchuck commented on app/helpers/users_helper.rb:
```diff
@@ -1,2 +1,9 @@
-end
+
+  # Returns the Gravatar for the given user.
+  def gravatar_for(user, size: 80)
```
make this function Single responsible. (it's doing more than one task)

chalchuck commented on app/views/users/new.html.erb:
```diff
@@ -1,3 +1,24 @@
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@user, url: signup_path) do |f| %>
```
I think you don't have to supply the `url` option here. 
This is because Rails is clever enough to figure out the path for @user if you've done routing as appropriate.

chalchuck commented on app/views/users/show.html.erb:
```diff
@@ -0,0 +1,11 @@
+  <aside class="col-md-4">
+    <section class="user_info">
+      <h1>
+        <%= gravatar_for @user %>
```
Personally I prefer having parameters wrapped in brackets.... (But it's okay if do not feel the same)

```suggestion
        <%= gravatar_for(@user) %>
```

chalchuck commented on config/puma.rb:
```diff
@@ -1,34 +1,16 @@
-# Allow puma to be restarted by `rails restart` command.
-plugin :tmp_restart
+on_worker_boot do
+  # Worker specific setup for Rails 4.1+
```
Please remove this comments

chalchuck commented on config/routes.rb:
```diff
@@ -4,4 +4,6 @@
   get  '/about',   to: 'static_pages#about'
   get  '/contact', to: 'static_pages#contact'
   get  '/signup',  to: 'users#new'
```
Use the rails way routes for full blown resources ....

`resources :person, only: []`

chalchuck replied:
if you want to make it a named route, then you can use the format you have used above. (If I am unclear, Please tell me so that I can explain further)

llabake replied:
Is this what you mean ```resources :users, only: [:new, :create]``` as opposed to having the named routes and the resources? @chalchuck

chalchuck replied:
yes but as time goes, you will need to remove the only option because you will be needing the other actions.

chalchuck commented on config/routes.rb:
```diff
@@ -4,4 +4,6 @@
   get  '/contact', to: 'static_pages#contact'
   get  '/signup',  to: 'users#new'
+  post '/signup',  to: 'users#create'
+  resources :users
```
you are having double routes for user. 

if you are using resources; then `get  '/signup',  to: 'users#new, post '/signup',  to: 'users#create'` are already created by the Rails Router

Please run `rails routes` to see your routes {focus on the user routes}
To get routes for a specific resource use `rails routes -c {controller name}` example `rails routes -c users`

chalchuck reviewed (commented):
Please review the following and submit.

llabake committed:
make redirect and flash massage a one liner
make the gravatar_url single responsibilty
remove comment in puma
remove redundant user_url in new.html.erb

chalchuck reviewed (approved):
Good to merge!
